tests/unit/utils/test_process.py

```diff
@@ -236,35 +236,158 @@ def test_daemonize_if(self):
+        'Validate MultiprocessingProcess and sub classes call after fork methods'
+        evt = multiprocessing.Event()
+        sig_to_mock = 'salt.utils.process.SignalHandlingMultiprocessingProcess._setup_signals'
+        log_to_mock = 'salt.utils.process.MultiprocessingProcess._setup_process_logging'
+        with patch(sig_to_mock) as ma, patch(log_to_mock) as mb:
+            self.sh_proc = salt.utils.process.SignalHandlingMultiprocessingProcess(target=self.no_op_target)
+            self.sh_proc._run()
+        ma.assert_called()
+        mb.assert_called()
+
+    @skipIf(NO_MOCK, NO_MOCK_REASON)
+    def test_signal_processing_test_final_methods_called(self):
+        'Validate MultiprocessingProcess and sub classes call finalize methods'
+        evt = multiprocessing.Event()
+        teardown_to_mock = 'salt.log.setup.shutdown_multiprocessing_logging'
+        log_to_mock = 'salt.utils.process.MultiprocessingProcess._setup_process_logging'
+        with patch(teardown_to_mock) as ma, patch(log_to_mock) as mb:
+            self.sh_proc = salt.utils.process.SignalHandlingMultiprocessingProcess(target=self.no_op_target)
+            self.sh_proc._run()
+        ma.assert_called()
+        mb.assert_called()
+
+    @staticmethod
+    def pid_setting_target(sub_target, val, evt):
+        val.value = os.getpid()
+        p = multiprocessing.Process(target=sub_target, args=(evt,))
+        p.start()
+        p.join()
+
+    @skipIf(sys.platform.startswith('win'), 'Required signals not supported on windows')
+    def test_signal_processing_handle_signals_called(self):
+        'Validate SignalHandlingMultiprocessingProcess handles signals'
+        # Gloobal event to stop all processes we're creating
+        evt = multiprocessing.Event()
+
+        # Create a process to test signal handler
+        val = multiprocessing.Value('i', 0)
+        proc = salt.utils.process.SignalHandlingMultiprocessingProcess(
+            target=self.pid_setting_target,
+            args=(self.run_forever_sub_target, val, evt),
+        )
+        proc.start()
+
+        # Create a second process that should not respond to SIGINT or SIGTERM
+        proc2 = multiprocessing.Process(
+            target=self.run_forever_target,
+            args=(self.run_forever_sub_target, evt),
+        )
+        proc2.start()
+
+        # Wait for the sub process to set it's pid
+        while not val.value:
+            time.sleep(.3)
+
+        assert not proc.signal_handled()
+
+        # Send a signal that should get handled by the subprocess
+        os.kill(val.value, signal.SIGTERM)
+
+        # wait up to 10 seconds for signal handler:
+        start = time.time()
+        while time.time() - start < 10:
+            if proc.signal_handled():
+                break
+            time.sleep(.3)
+
+        assert proc.signal_handled()
+        # Reap the signaled process
+        proc.join(1)
+        assert psutil.pid_exists(proc2.pid)
```

Comment:
If the assert fails the event set and proc join won't run...

Reply:
Safe assertion?

Reply:
Yeah, something is weird since we'd never reap proc2. I will fix.